### PR TITLE
refactor: 💡 Refactor `sessions-list` to use `Hds::Table`

### PR DESCRIPTION
**Pull request by dvonanderson**
✅ Closes: https://hashicorp.atlassian.net/browse/ICU-11732

:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-11732)

## Description
- Refactor `sessions-list` to use `Hds::Table` instead of `Rose::table`

### Before: 
---

Screenshot 2023-12-13 at 10 45 36

### After:
---
Screenshot 2023-12-13 at 10 54 41

## How to Test
- [ ] Click on Vercel link for Admin UI or pull down PR and run it locally using `yarn start`
- [ ] Click on any "Orgs" card created by mirage
- [ ] Click on any "Projects" card created by mirage and "Sessions" will appear in the sidebar navigation.
- [ ] Click on "Sessions" in the sidebar navigation to view refactored sessions-list

:technologist: [Admin preview](https://boundary-ui-git-icu-11732-admin-ui-use-hds-tab-c933e2-hashicorp.vercel.app/)

## Checklist:

- [x] I have added before and after screenshots for UI changes
~~- [ ] I have added JSON response output for API changes~~
~~- [ ] I have added steps to reproduce and test for bug fixes in the description~~
~~- [ ] I have commented on my code, particularly in hard-to-understand areas~~
- [x] My changes generate no new warnings
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~

### CODE REVIEW

File: ui/admin/app/styles/app.scss [hunk 1]
```diff
@@ -740,7 +740,8 @@
 }
 
 .auth-methods-list-table,
-.workers-list-table {
+.workers-list-table,
+.sessions-list-table {
```
Comment (dvonanderson) [inline ui/admin/app/styles/app.scss:744]: Do we want to create a more generic class name for this and spread it to all of the tables that need?

Comment (calcaide) [reply ui/admin/app/styles/app.scss:744]: In previous HDS work I left this comment: "If this specific change is causing friction or raising doubts, I will suggest to stick the PR to rafctor the HDS::Table and bring the toolbar issue to tech time. Toolbar is used in different places of our UI, so can be good to dedicate a specific ticket/PR to "normalize" the toolbar."

My opinion here is: (feel free to discuss or NOT agree)
- We have lots of tables.
- We have some of those tables also using a toolbar component.
- Let's work and finish first the tables. That will give us a better understanding of the toolbar "issue".
- Once we have better understanding, we can tackle it in a specific task and PR where the topic is just the toolbar.

File: ui/admin/app/templates/scopes/scope/sessions/index.hbs [hunk 2]
```diff
@@ -68,76 +68,64 @@
...
-                    {{t 'actions.cancel'}}
-                  </Rose::Button>
-                {{/if}}
-              </row.cell>
-            </body.row>
-          {{/each}}
-        </table.body>
-      </Rose::Table>
+                  {{t 'actions.cancel'}}
+                </Rose::Button>
```
[the remark this answers is not in the document]
Comment (dvonanderson) [reply ui/admin/app/templates/scopes/scope/sessions/index.hbs:123]: @cameronperera yes good catch, will do!

File: ui/admin/app/templates/scopes/scope/sessions/index.hbs [hunk 3]
```diff
@@ -68,76 +68,64 @@
...
+                class={{concat 'session-status-' B.data.session.status}}
+                @name='flight-icons/svg/entry-point-16'
+                @size='medium'
+              />
+              <Copyable
+                @text={{B.data.session.id}}
+                @buttonText={{t 'actions.copy-to-clipboard'}}
+                @acknowledgeText={{t 'states.copied'}}
+              >
+                <Hds::Text::Code>{{B.data.session.id}}</Hds::Text::Code>
```
Comment (lisbet-alvarez) [inline ui/admin/app/templates/scopes/scope/sessions/index.hbs:97]: nice! i didn't know there was a new hds code component!

File: ui/admin/app/templates/scopes/scope/sessions/index.hbs [hunk 4]
```diff
@@ -68,76 +68,64 @@
...
+            <B.Td>
+              <time datetime={{format-date-iso B.data.session.created_time}}>
+                {{format-date-iso-human B.data.session.created_time}}
+              </time>
+            </B.Td>
+            <B.Td>
+              <SessionStatus @model={{B.data.session}} />
+            </B.Td>
+            <B.Td>
+              {{#if B.data.session.isAvailable}}
```
Comment (lisbet-alvarez) [inline ui/admin/app/templates/scopes/scope/sessions/index.hbs:117]: This should check the `can 'cancel session'` ability (which checks if the session is available and if the session has `cancel` authorized action)

File: ui/admin/app/styles/app.scss (hunk 1 above)
Comment (dvonanderson) [reply ui/admin/app/styles/app.scss:744]: @calcaide @cameronperera thank you both for the information!  Based on this conversation and the one from tech-time, I will just use the custom CSS.

File: ui/admin/app/templates/scopes/scope/sessions/index.hbs (hunk 3 above)
Comment (dvonanderson) [reply ui/admin/app/templates/scopes/scope/sessions/index.hbs:97]: @lisbet-alvarez yea pretty nifty!

File: ui/admin/app/templates/scopes/scope/sessions/index.hbs (hunk 4 above)
Comment (dvonanderson) [reply ui/admin/app/templates/scopes/scope/sessions/index.hbs:117]: @lisbet-alvarez is that in reference to the action to the `route-action` to cancel a session?

Comment (lisbet-alvarez) [reply ui/admin/app/templates/scopes/scope/sessions/index.hbs:117]: i'm thinking the if statement on line 117 should be replaced with `{{#if (can 'cancel session' B.data.session)}}` so the cancel session button is only displayed when they have cancel action. Or is there a reason we don't want to check cancel permission?

Comment (dvonanderson) [reply ui/admin/app/templates/scopes/scope/sessions/index.hbs:117]: @lisbet-alvarez yes good point.  We already check for the availability in the `SessionAbility'

File: ui/admin/app/templates/scopes/scope/sessions/index.hbs [hunk 5]
```diff
@@ -68,76 +68,64 @@
...
+        <:body as |B|>
+          <B.Tr>
+            <B.Td>
+              <Rose::Icon
+                class={{concat 'session-status-' B.data.session.status}}
+                @name='flight-icons/svg/entry-point-16'
+                @size='medium'
+              />
+              <Copyable
+                @text={{B.data1session.id}}
```
[the remark this answers is not in the document]
Comment (dvonanderson) [reply ui/admin/app/templates/scopes/scope/sessions/index.hbs]: @cameronperera Thank you so much for the update.

**Review by cameronperera — APPROVED** [review]

**Review by lisbet-alvarez — APPROVED** [review]
LGTM!

**Pull request merged**